src/hatch_rest_api/util_bootstrap.py

```diff
@@ -76,6 +76,7 @@ def create_rest_devices(iot_device):
                 thing_name=iot_device["thingName"],
                 mac=iot_device["macAddress"],
                 shadow_client=shadow_client,
+                sounds=sounds,
```

Comment:
Based on how `_get_sound_content_for_v2_devices` is defined right now, it won't be. API calls to Hatch are only made for `if device["product"] in ["riot", "riotPlus"]`, otherwise an empty list is returned.

I'd personally like to keep this (and below) in the code, as I have some subsequent changes for some refactorings to clean up the code and make things more consistent across different devices that I'd like to put up for review (after (a) getting this change merged, (b) consumed by the HA component, and (c) tested for a sufficient enough period of time we're happy/convinced the changes are good - specifically, because Hatch does not appear to have API documentation, at least to the best of my knowledge, I feel like changes to a lib like this need some decent real-world testing for building adequate confidence).

Given that, let me know if this is still a concern for you!

Reply:
The main reason we've kept the gen1 vs gen2 code as cleanly separate as possible is because people have only come forward to test when they experience a problem.  In the ha issue associated with this pr, I've seen multiple users willing to test from the gen2 perspective, but nobody from the gen1 side.  This is typical and for user stability we should continue avoiding intermixing gen1/gen2 code outside of the bootstrapping.

Ideally the common init shouldn't have been allowed to include a gen2 param, you could correct that in this pr or I can in a few weeks when I get back to the States from Japan.

Gen1 already has a working sound mode list that is implemented very differently.

Reply:
Gotcha - for ease, I'll just remove this (and below) for now.

Perhaps a bit unrelated (but, confusing to me none-the-less), it seems that, at least based on this code base, there is some nuanced differences anyways between gen2 devices, since the lists of products in the `*_v2_*` methods check against different lists:
![Screenshot 2024-04-04 at 23 20 37](https://github.com/dahlb/hatch_rest_api/assets/6374067/4f5720b5-cc4d-447a-896c-bbfa4a31dc25)

I've pushed a change to resolve your initial concerns, but I'd love to know (mostly for subsequent knowledge/changes I might put forward) if the above is expected/true (that is, are there differences in the gen2 devices such that these product checks differ, or is this unintentional/accidental)?

Reply:
yeah my guess is hatch employeed separate 3rd party contractor teams for gen1 vs gen2.  The API urls, naming schemas, and response formats are completely different.  The only overlap is some of the auth, which is a large part of why all the code has been kept separate.  The initial pr to add gen2 tried to use the same classes perticularly in ha and that caused a ton of branching in every method so even ha's code base tries to keep the code separate except for the bootstrapping where the auth overlaps.